### PR TITLE
Add -Lv/--line-verbose mode to rdump

### PR DESCRIPTION
**Pull request by yunzheng**
This adds a verbose option to the LineWriter adapter which also prints the field type.

### CODE REVIEW

File: flow/record/adapter/line.py [hunk 1]
```diff
@@ -15,23 +18,35 @@ class LineWriter(AbstractWriter):
...
         if isinstance(self.fields, str):
             self.fields = self.fields.split(",")
         if isinstance(self.exclude, str):
             self.exclude = self.exclude.split(",")
 
     def write(self, rec):
         rdict = rec._asdict(fields=self.fields, exclude=self.exclude)
+        rdict_types = None
+        if self.verbose:
+            rdict_types = {fname: fieldset.typename for fname, fieldset in rec._desc.get_all_fields().items()}
```
Comment (Schamper) [inline flow/record/adapter/line.py:36]: You could maybe `lru_cache` this on `rec._desc`.

**Review by Schamper — CHANGES_REQUESTED** [review]

File: flow/record/adapter/line.py (hunk 1 above)
Comment (yunzheng) [reply flow/record/adapter/line.py]: nice one, added! including a test to ensure the caching works correctly.

File: flow/record/adapter/line.py [hunk 2]
```diff
@@ -1,37 +1,66 @@
...
+Write usage: rdump -w line://[PATH]?verbose=[VERBOSE]
 [PATH]: path to file. Leave empty or "-" to output to stdout
+
+Optional arguments:
+    [VERBOSE]: Also show fieldtype in line output (default: False)
 """
 
 
+@lru_cache(maxsize=1024)
+def field_types_for_record_descriptor(desc: RecordDescriptor) -> dict[str, str]:
```
Comment (Schamper) [inline flow/record/adapter/line.py:21]: Maybe nice to also add annotations to the rest of the file?

Comment (yunzheng) [reply flow/record/adapter/line.py:21]: Done

**Review by Schamper — APPROVED** [review]

**Pull request merged**